lib/scholar/cluster/gmm.ex

```diff
@@ -0,0 +1,407 @@
+defmodule Scholar.Cluster.GaussianMixture do
+  @moduledoc """
+  Gaussian Mixture Model.
+
+  Gaussian Mixture model is a probabilistic model that assumes every data point is generated
+  by randomly choosing one of the several fixed Gaussian distributions and then sampling from it.
+  Its parameters are estimated using the Expectation-Maximization (EM) algorithm, which is an
+  iterative algorithm alternating between the two steps: the E-step which computes the
+  expectation of the Gaussian assignment for each data point x and the M-step which updates the
+  parameters to maximize the expectations found in E-step. While every iteration of the algorithm
+  is guaranteed to improve the log-likelihood, the final result depends on the initial values of
+  the parameters and the entire procedure should be repeated several times.
+
+  References:
+
+  * [1] - Mixtures of Gaussians and the EM algorithm https://cs229.stanford.edu/notes2020spring/cs229-notes7b.pdf
+  """
+
+  import Nx.Defn
+  import Scholar.Shared
+  require Nx
+
+  @derive {Nx.Container, containers: [:weights, :means, :covariances, :precisions_cholesky]}
+  defstruct [:weights, :means, :covariances, :precisions_cholesky]
+
+  opts = [
+    num_gaussians: [
+      required: true,
+      type: :pos_integer,
+      doc: "The number of Gaussian components in the mixture."
```

Comment:
```suggestion
      doc: "The number of Gaussian model components in the mixture."
```

Reply:
The Gaussian here refers to the Gaussian distribution. Perhaps writing "The number of Gaussian distributions in the mixture" would be better.